src/adapter/file_adapter.rs

```diff
@@ -28,13 +28,16 @@ use tokio::{
 
 use async_trait::async_trait;
 
-pub struct FileAdapter<P: AsRef<Path>> {
+pub struct FileAdapter<P> {
     pub file_path: P,
 }
```

Comment:
If we removed the `pub` field so it can only be constructed via `new` this is more common with how stdlib works only put bounds on the impl blocks. I have no problem reverting this back to original setup.